resources/assets/js/annotations/components/annotationCanvas/drawInteractions.vue

```diff
@@ -101,29 +101,25 @@ export default {
             }
         },
         interactionMode(mode) {
-            if (this.canAdd) {
-                this.maybeUpdateDrawInteractionMode(mode)
-            }
+            this.maybeUpdateDrawInteractionMode(mode)
+
         },
-        canAdd(state) {
-            if (state) {
-                Keyboard.on('a', this.drawPoint, 0, this.listenerSet);
-                Keyboard.on('s', this.drawRectangle, 0, this.listenerSet);
-                Keyboard.on('d', this.drawCircle, 0, this.listenerSet);
-                Keyboard.on('Shift+d', this.drawEllipse, 0, this.listenerSet);
-                Keyboard.on('f', this.drawLineString, 0, this.listenerSet);
-                Keyboard.on('g', this.drawPolygon, 0, this.listenerSet);
-            }
-            else {
-                drawInteraction?.setActive(false);
-                Keyboard.off('a', this.drawPoint, 0, this.listenerSet);
-                Keyboard.off('s', this.drawRectangle, 0, this.listenerSet);
-                Keyboard.off('d', this.drawCircle, 0, this.listenerSet);
-                Keyboard.off('Shift+d', this.drawEllipse, 0, this.listenerSet);
-                Keyboard.off('f', this.drawLineString, 0, this.listenerSet);
-                Keyboard.off('g', this.drawPolygon, 0, this.listenerSet);
-            }
+        canAdd: {
+            handler(state) {
+                if (!state) {
+                    this.resetInteractionMode();
+                }
+            },
+            immediate: true
```

Comment:
This watcher would be required for all the other interactions, too (e.g. the `attachlabelInteraction`). However, the reset would be called many times then, even though it's required only once. So what about moving this into the main file `annotationCanvas.vue`? It could just be called for `canAdd`, `canModify` and `canDelete`. These don't need to be immediate watchers, too (since the initial state is the default interaction mode).